Makefile

```diff
@@ -9,38 +9,34 @@
 SIL ?= @
 MAKEFLAGS += --no-print-directory
 
-
-#TARGET ?= ia32-generic
-#TARGET ?= armv7m3-stm32l152xd
-#TARGET ?= armv7m3-stm32l152xe
-#TARGET ?= armv7m4-stm32l4x6
-#TARGET ?= armv7m7-imxrt105x
-#TARGET ?= armv7m7-imxrt106x
-#TARGET ?= armv7m7-imxrt117x
-#TARGET ?= armv7a7-imx6ull
-TARGET ?= ia32-generic
-#TARGET ?= riscv64-spike
-#TARGET ?= host-pc
-
 include ../phoenix-rtos-build/Makefile.common
+# FIXME: this include should be done by Makefile.common
 include ../phoenix-rtos-build/Makefile.$(TARGET_SUFF)
 
 CFLAGS += $(BOARD_CONFIG)
 
+# FIXME: to be moved to Makefile.common - paths to the template files
+binary.mk := binary.mk
+static-lib.mk := static-lib.mk
 
-.PHONY: clean
-clean:
-	@echo "rm -rf $(BUILD_DIR)"
+# function - returns the directory of the current Makefile
+my-dir = $(dir $(lastword $(MAKEFILE_LIST)))
 
-ifneq ($(filter clean,$(MAKECMDGOALS)),)
-	$(shell rm -rf $(BUILD_DIR))
-endif
+.DEFAULT_GOAL := all
 
-T1 := $(filter-out clean all,$(MAKECMDGOALS))
-ifneq ($(T1),)
-	include $(T1)/Makefile
-.PHONY: $(T1)
-$(T1): all
-else
-	include _targets/Makefile.$(TARGET_FAMILY)-$(TARGET_SUBFAMILY)
+ifneq ($(filter %clean,$(MAKECMDGOALS)),)
+.NOTPARALLEL:
 endif
+
+# should define DEFAULT_COMPONENTS and target-specific variables
+include _targets/Makefile.$(TARGET_FAMILY)-$(TARGET_SUBFAMILY)
+
+# read out all components
+ALL_MAKES := $(wildcard */*/Makefile) $(wildcard */*/*/Makefile)
```

Comment:
Wouldn't it be better to use the `find` command :
```
ALL_MAKES := $(filter-out ./Makefile,$(shell find -name Makefile))
```

Reply:
Yeah, builtin should be quicker.

IMHO this is a minor issue, I went wit the easiest approach and will see if it would work in other repos.

I don't know if including "child" Makefiles by other Makefiles is the way to go - if somebody would reintroduce `*/*/*/Makefile` here (or go with `find` approach) we will be including one makefile multiple times.